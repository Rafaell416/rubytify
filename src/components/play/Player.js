import React from 'react'
import PropTypes from 'prop-types'
import { Heading2, Body } from '../typography' 
import Button from '../common/buttons/Button'

function Player ({ song: { name, preview_url, spotify_url } }) {
  return (
    <div style={style.container}>
      <Heading2>{name}</Heading2>
      { preview_url 
        ? <video style={{ outline: 'none' }} controls autoPlay name="media">
            <source src={preview_url} type="audio/mpeg"/>
          </video>
        : <Body>Loading...</Body>
      }
      <div style={style.buttonContainer}>
        <Button outline>
          <a href={spotify_url} target="_blank" style={style.a}>Play on Spotify</a>
        </Button>
      </div>
    </div>
  )
}

Player.propTypes = {
  name: PropTypes.string,
  preview_url: PropTypes.string,
  spotify_url: PropTypes.string
}

const style = {
  container: {
    display: 'flex',
    padding: 50,
    flexDirection: 'column',
    backgroundColor: 'transparent',
    alignItems: 'center',
    justifyContent: 'center'
  },
  a: {
    color: 'white'
  },
  buttonContainer: {
    marginTop: 20
  }
}

export default Player