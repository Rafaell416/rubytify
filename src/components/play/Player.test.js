import React from 'react'
import Player from './Player'
import renderer from 'react-test-renderer'

const song = {
  id: 1,
  name: 'testing song',
  preview_url: 'https://cdn.test.mp3',
  spotify_url:'https://open.spotify.com/song'
}

test('Testing <Player /> component', () => {
  const component = renderer.create(
    <Player song={song}/>
  )
  const instance = component.root

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
  expect(instance.children[0]._fiber.elementType).toBe('div')
})