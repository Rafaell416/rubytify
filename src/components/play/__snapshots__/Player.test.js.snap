// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <Player /> component 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "transparent",
      "display": "flex",
      "flexDirection": "column",
      "justifyContent": "center",
      "padding": 50,
    }
  }
>
  <h2
    style={
      Object {
        "color": "white",
        "fontFamily": "Roboto, sans-serif",
        "margin": 0,
      }
    }
  >
    testing song
  </h2>
  <video
    autoPlay={true}
    controls={true}
    name="media"
    style={
      Object {
        "outline": "none",
      }
    }
  >
    <source
      src="https://cdn.test.mp3"
      type="audio/mpeg"
    />
  </video>
  <div
    style={
      Object {
        "marginTop": 20,
      }
    }
  >
    <button
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "border": ".5px solid",
          "borderColor": "#2EE59D",
          "borderRadius": 25,
          "cursor": "pointer",
          "display": "flex",
          "height": 40,
          "justifyContent": "center",
          "marginLeft": 5,
          "marginRight": 5,
          "outline": "none",
          "width": 150,
        }
      }
    >
      <h2
        style={
          Object {
            "color": "white",
            "fontFamily": "Roboto, sans-serif",
            "margin": 0,
          }
        }
      >
        <a
          href="https://open.spotify.com/song"
          rel="noopener noreferrer"
          style={
            Object {
              "color": "white",
            }
          }
          target="_blank"
        >
          Play on Spotify
        </a>
      </h2>
    </button>
  </div>
</div>
`;
